### ISSUE TITLE
Kitty image preview freezing ranger when using commands

### ISSUE THREAD

**Issue by PoisonFrog** [kovidgoyal/kitty#3864]
Whenever I have an image being previewed in ranger and I use a shell command inside ranger the app freezes and I have to close the terminal window. Terminal keeps working but the ranger window becomes totally frozen and unusable.

1. Open ranger with kitty image preview enabled
2. Find an image and select it
3. Type a commands like "ls"

I have kitty 0.21.2 installed on Fedora 34. I tried it in Ubuntu 20.04 and it also froze. A guy in the ranger github page said he could not reproduce the bug and no one knew how to help, maybe it has something to do with kitty idk (other image previews protocols are working fine in ranger).

**Comment by kovidgoyal** [comment]
This is a ranger bug. It is writing invalid escape codes to the
terminal. Easily checked by starting kitty from another terminal and
then trigerring the issue kitty will print
[PARSE ERROR] Unknown char after ESC: 0x5c

indicating invalid escape codes being sent to it.

**Comment by kovidgoyal** [comment]
And for further clarification, 0x5c is the backslash character. kitty is saying ranger sent it an unexpected <ESC><backslash> this sequence is called a "string terminator" and is sent to terminate the graphics escape code. It it being received out of sequence. 

And kitty is not freezing ranger is, also easily verified by doing a pkill ranger, after which the kitty window is active again.

**Comment by PoisonFrog** [comment]
> This is a ranger bug. It is writing invalid escape codes to the
> terminal. Easily checked by starting kitty from another terminal and
> then trigerring the issue kitty will print
> [PARSE ERROR] Unknown char after ESC: 0x5c
> 
> indicating invalid escape codes being sent to it.

Are you sure? When I try it with w3m image preview method it outputs the same codes but ranger doesn't freeze.

**Comment by kovidgoyal** [comment]
On Sun, Jul 25, 2021 at 08:19:46AM -0700, PoisonFrog wrote:
> > This is a ranger bug. It is writing invalid escape codes to the
> > terminal. Easily checked by starting kitty from another terminal and
> > then trigerring the issue kitty will print
> > [PARSE ERROR] Unknown char after ESC: 0x5c
> > 
> > indicating invalid escape codes being sent to it.
> 
> Are you sure? When I try it with w3m image preview method it outputs the same codes but ranger doesn't freeze.

I have no clue what the issue in ranger is. Those errors indicate
ranger is doing *something* wrong. And the fact that it is
freezing is another strong indication. Most likely because it is waiting
for some response from the terminal that it will never get because of a
previously malformed escape code. I am not a ranger developer, I am not
going to debug it for you.

**Comment by PoisonFrog** [comment]
> On Sun, Jul 25, 2021 at 08:19:46AM -0700, PoisonFrog wrote: > This is a ranger bug. It is writing invalid escape codes to the > terminal. Easily checked by starting kitty from another terminal and > then trigerring the issue kitty will print > [PARSE ERROR] Unknown char after ESC: 0x5c > > indicating invalid escape codes being sent to it. Are you sure? When I try it with w3m image preview method it outputs the same codes but ranger doesn't freeze.
> I have no clue what the issue in ranger is. Those errors indicate ranger is doing *something* wrong. And the fact that it is freezing is another strong indication. Most likely because it is waiting for some response from the terminal that it will never get because of a previously malformed escape code. I am not a ranger developer, I am not going to debug it for you.

I realized something. You said the error message had something to do with backspace and I just remembered that when I switched to Kitty I began having problems when connecting to Linux servers via SSH because my backspace wasn't working properly. Then I searched online for a solution and someone told me to use these commands:
```
stty sane
export TERM=linux
```
then it behaved as expected but I only had this problem with Kitty. Till this day I have these commands in the /etc/profile file on these servers. Maybe these problems are related?

When I use other terminals to access ranger I don't get these parse errors. I also get similar error messages with ncmpcpp and these also only happens with Kitty even though I can't notice anything going wrong when using ncmpcpp. It seems that Kitty doesn't behave the same way as native linux terminals do, but I don't know. I am not a programmer or anything, it is just speculation.

**Comment by kovidgoyal** [comment]
DO NOT change the value of TERM. Set it back to xterm-kitty. And if you
have problems sshing into servers, read
https://sw.kovidgoyal.net/kitty/faq/#i-get-errors-about-the-terminal-being-unknown-or-opening-the-terminal-failing-when-sshing-into-a-different-computer

**Comment by Luflosi** [comment]
You should never set `TERM` to a different value or various things will break. See https://github.com/kovidgoyal/kitty/issues/434#issuecomment-767666672 for a more detailed explanation. Also see https://sw.kovidgoyal.net/kitty/faq/#i-get-errors-about-the-terminal-being-unknown-or-opening-the-terminal-failing-when-sshing-into-a-different-computer.
TLDR: set TERM correctly and install the xterm-kitty file in the correct place (can be done by installing the `kitty-terminfo` package with the package manager on most distros).

**Comment by PoisonFrog** [comment]
I didn't change the value of TERM in my system. The value is still xterm-kitty.

**Comment by kovidgoyal** [comment]
Indeed, this issue has nothing to do with TERM, its a bug in ranger. And I mentioned backSLASH not backSPACE. I am simply saying, DO NOT change TERM on your servers.
